E3SM-Project/esgfpub
# Refactor repository scaffolding

## Overview
This PR updates the scaffolding of the repository to follow Python conventions. It also removes the `esgfpub` directory since it is no longer being used.

## Summary of changes
- Move `/warehouse` directory one level down to the root of the repo.
  - This directory was previously `/warehouse/warehouse`.
  - Python module imports were safely updated using PyCharm.
- Rename `/warehouse` directory  to `/datasm`.
  - Python module imports were safely updated using PyCharm.
  - Update some `warehouse` string refs to `datasm`, except paths, command arguments, and classes.
- Remove `/esgfpub` directory and related refs.
  - Closes #104 

DevOps
- Move `/datasm/setup.py` to the root of repo and update contents.
  - Now we run `pip install . ` from the root of the repo, instead of `cd warehouse` first.
- Rename `dev.yml` to `proc.yml`. 
  - Update references for "development" to "processing"
  - It is more accurate to call this environment the "processing" env because the "publishing" env is also used for development.
- Remove `prod.yml` since it is not working yet. 
   - Need to resolve #110 and #99 first

Documentation
- Update `README.md` and `docs/getting_started.md`
- Move `NewCMIPVariableSetup.md` and `warehouse_processing.md` to `/docs`
- Update `.gitignore` with standard list from https://github.com/github/gitignore/blob/main/Python.gitignore

## What to review in this PR
I already performed a self-review of the items below, but more eyes are always. Most of the changes in the diff are related to deleting the `/esgfpub` directory, so you can ignore those

- [x] Make sure documentation updates are free of error (mostly path updates)
- [x] Local `datasm` build still builds and installs in the processing and publishing envs
- [x] Local `datasm` build runs the same as before using `datasm` command, instead of `warehouse`
  - Example: `datasm --help`, `datasm validate --help`

## How to test this PR

Processing operations -- `extract`, `validate`, `postprocess`
1. Install and activate `datasm_proc` env
   - `conda env install -f conda-env/proc.yml`
   - `conda activate datasm_proc`
2. Install local build into `datasm_proc` env
   - Make sure you are in the root of `esgfpub`
   - `pip install .`
3. Run `datasm` processing operations to make sure things are good to go
   - For example, confirm there are no `ModuleNotFound` errors

Publishing operations -- `process`
1. Install and activate `datasm_pub` env
   - `conda env install -f conda-env/pub.yml`
   - `conda activate datasm_pub`
2. Install local build into `datasm_pub` env
   - Make sure you are in the root of `esgfpub`
   - `pip install .`
3. Run `datasm` publishing operation to make sure things are good to go
   - For example, confirm there are no `ModuleNotFound` errors

## Next Actions
- [x] Rename the repository to `datasm`
